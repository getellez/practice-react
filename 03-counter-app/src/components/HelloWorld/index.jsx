import React from 'react'

export const HelloWorld = () => {
  const newMessage = {
    author: 'German',
    date: new Date(),
    categories: [1,2,3,4,5]
  }
  const getResult = () => {
    return 2+2;
  }
  return (
    <>
      <h1>Este es mi primer componente</h1>
    </>
  )
}
