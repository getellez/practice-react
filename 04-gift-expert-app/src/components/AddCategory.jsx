import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ handleAddCategory }) => {
  const [inputValue, setInputValue] = useState('')
  
  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }
  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return
    handleAddCategory(inputValue.trim())
    setInputValue('')
  }
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text" 
        placeholder="Buscar GIFs"
        name="categoryInput"
        autoComplete="off"
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
  handleAddCategory: PropTypes.func.isRequired,
}