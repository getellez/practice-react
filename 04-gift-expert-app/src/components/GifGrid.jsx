import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifItem } from "./GifItem"
import PropTypes from 'prop-types'

export const GifGrid = ({ category, handleRemoveCategory }) => {
  
  const { images, isLoading } = useFetchGifs(category)
  
  return (
    <>
      <h3> { category } </h3> 
      {/* <small>[Eliminar]</small> */}
      <small onClick={() => handleRemoveCategory(category)}>[Eliminar]</small>
      
      {
        isLoading && (<h2>Loading...</h2>)
      }
      <div className="card-grid">
        {
          images.map(({ url, title, id }) => (
            <GifItem key={id} url={url} title={title} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
  handleRemoveCategory: PropTypes.func.isRequired
}