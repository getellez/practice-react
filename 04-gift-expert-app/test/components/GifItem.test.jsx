import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('<GifItem /> component', () => {
  const title = 'Saitama'
  const url = 'https://www.test.com/'
  
  test('debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url}/>)
    expect(container).toMatchSnapshot()
  })

  test('should show the image with the URL and ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(src)
    expect(alt).toBe(title)
  })

  test('should show tht title in the component', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })

})