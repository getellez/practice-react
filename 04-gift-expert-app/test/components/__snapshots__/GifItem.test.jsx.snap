// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> component debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://www.test.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
